### DISCUSSION TITLE
FluxImg2ImgPipeline on Flux-1 Schnell image to image output troubles

### DISCUSSION THREAD

**Discussion by Dannyvannnn** [huggingface/diffusers#9898] [Q&A]
Hi im trying to use the image to image pipeline for Flux-1 Schnell to do corrections on images (e.g changing clothes color, hair color, eye color etc.) However it seems like Flux-1 schnell cannot properly do such a thing no matter what prompts i tried. I have tried other models such as SD2 or SDXL and they manage to make the corrections on the image, even if the quality of the output may not be very good.  I want to ask how can i edit the code/parameters/prompting style/etc. in order to achieve my requirements.

First image of each row is the input image, difference between first screenshot and second screenshot is increasing strength from 0.7 to 0.8, which appears to be the threshold for the change. Code is copied from the FluxPipeline page on HuggingFace
![Screenshot_794](https://github.com/user-attachments/assets/f96bf094-d70b-4630-8720-af3fb57925c6)
![Screenshot_793](https://github.com/user-attachments/assets/1d226cd3-d478-4e7a-a2d6-accb4d232546)

**Comment by ukaprch** [comment]
Have you considered using the inpaint pipeline masking the areas you wish to change or fill? Then your prompt would be more successful in achieving your goals.

**Reply by Dannyvannnn** [reply]
Hi! yes i've considered using the inpaint pipeline for this project if i want more accurate results, However i would like to test the Img2Img pipeline and its limits and possible use cases as inpainting takes more work to integrate into a webapp project. So far I am unsure on how to properly use the Img2Img pipeline as from my testing, it either follows the input image too strictly, or it does not follow the image at all.

Here is a clearer example of what i mean. 

My Expectation: Even without specifying the desired art style (anime style), the pipeline should generate an anime art style image with the due to the input image. At lower strength levels, the model should follow the base image (e.g changing the orange sweater of the image to a green sweater). At higher strength levels, there can be more variety in for e.g poses or accessories, but there should still be an anime style girl in a green sweater.

Reality: At lower strength levels, my prompts appear to not do anything, the model only appears to make minor edits to the base image. At higher strength levels, it does not take my input image into consideration at all and outputs what it understands from the text prompt

Am i misunderstanding how the Img2Img pipeline works? is there a set of examples where i can properly see how the Flux pipeline's Img2Img is intended to work? 
![Screenshot_799](https://github.com/user-attachments/assets/08ee5f3d-d7f8-41d1-9b3c-dd9c78fa1def)
![Screenshot_798](https://github.com/user-attachments/assets/b228dd5c-21d9-4412-8c10-7e4bbe1c3278)

In comparison with SDXL-Turbo (also a timestep-distilled model similar to Flux-1 Schnell), the results of which can be seen below, even though the quality of the edit may not be good, I can still see attempts by the model to follow the prompt. 

![Screenshot_800](https://github.com/user-attachments/assets/c9bf5549-e678-4f59-8cb6-c1844529da3d)
![Screenshot_801](https://github.com/user-attachments/assets/3552dce4-dc6a-415a-81ab-f5beee401431)

Other closed-source models also are able to follow similar prompts with outputs closer to what i am imagining Flux's Img2Img should be able to output
![Screenshot_803](https://github.com/user-attachments/assets/f33c78ff-2aab-4dba-8185-d104e9a6b356)

**Comment by vgabbo** [comment]
Hi @Dannyvannnn , you should try using 4 steps and strength 0.75 (or 0.7499999). Or 10 steps and strength 0.9 ( or 0.899999). It helps in my case.
